src/Databases/DataLake/ICatalog.h

```diff
@@ -105,6 +107,16 @@ class TableMetadata
 };
 
 
+struct CatalogSettings
+{
+    String storage_endpoint;
+    String aws_access_key_id;
+    String aws_secret_access_key;
+    String region;
+
+    DB::SettingsChanges allChanged() const;
```

Comment:
I do not see this method being used anywhere, remove?

Reply:
it's useful here https://github.com/ClickHouse/ClickHouse/blob/master/src/Databases/DataLake/GlueCatalog.cpp#L442